### PR TITLE
12080 - Zooming in and out no longer loses track of player's preferred center point for map

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/module/Map.java [hunk 1]
```diff
@@ -3768,25 +3793,41 @@ public String getIdentifier() {
...
+    scroll.unregisterKeyboardAction(
+      KeyStroke.getKeyStroke(KeyEvent.VK_PAGE_UP, 0));
+    scroll.setAlignmentX(0.0f);
+    scroll.setAlignmentY(0.0f);
+
+    layeredPane.setLayout(new InsetLayout(layeredPane, scroll));
+    layeredPane.add(scroll, JLayeredPane.DEFAULT_LAYER);
+
+    // When our Viewport changes for any reason, check if we need to update our "preferred center point"
+    scroll.getViewport().addChangeListener(new ChangeListener() {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/Map.java:3815]: Change listeners can be replaced with lambdas:
```
scroll.getViewport().addChangeListener(e -> updateCenter());
```

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/Map.java]: fixed

File: vassal-app/src/main/java/VASSAL/build/GameModule.java [hunk 2]
```diff
@@ -337,6 +339,32 @@ public void setSuppressSounds(boolean suppressSounds) {
...
 
+
+
+
+  private final Timer zoomTimer = new Timer(1000, new ActionListener() {
+    @Override
+    public void actionPerformed(ActionEvent e) {
+      suppressAutoCenterUpdate = false;
+    }
+  });
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/GameModule.java:350]: The listener can be a lambda here:
```
private final Timer zoomTimer = new Timer(1000, e -> suppressAutoCenterUpdate = false);
```

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/GameModule.java]: Nifty spifty - fixed

**Pull request merged**